File: GFG/Recursion/generatesubsets.cpp
#include<iostream>
#include<string>
using namespace std;

void gensubset(string s,string curr="",int i=0){
    if (i==s.length()){
        cout<<curr<<" ";
        return ;
    }
    gensubset(s,curr,i+1);
    gensubset(s,curr+s[i],i+1);
}

int main (){
    string x;
    cin>>x;
    gensubset(x);
    return 0;
}

File: GFG/Recursion/generatesubsets.py
def gensubsets(s,curr="",i=0):
    if(i==len(s)):
        print(curr)
        return
    gensubsets(s,curr,i+1)
    gensubsets(s,curr+s[i],i+1)
    
x=input()
gensubsets(x)